src/common-utils/main.sh

```diff
@@ -408,10 +408,15 @@ fi
 if [ "${USERNAME}" = "root" ]; then
     user_home="/root"
 else
-    user_home="/home/${USERNAME}"
-    if [ ! -d "${user_home}" ]; then
-        mkdir -p "${user_home}"
-        chown ${USERNAME}:${group_name} "${user_home}"
+    # Check if user already has a home directory other than /home/${USERNAME}
```

Comment:
How about updating L:408 to LL421 with following code? 👇 (The logic is the same, just reducing the number of conditions to improve readability)

```
if [ "${USERNAME}" = "root" ]; then
    user_home="/root"
# Check if user already has a home directory other than /home/${USERNAME}
elif [ "/home/${USERNAME}" != $( getent passwd $USERNAME | cut -d: -f6 ) ]; then
    user_home=$( getent passwd $USERNAME | cut -d: -f6 )
else
    user_home="/home/${USERNAME}"
    if [ ! -d "${user_home}" ]; then
        mkdir -p "${user_home}"
        chown ${USERNAME}:${group_name} "${user_home}"
    fi
fi
```

Reply:
thanks! I've added the suggestion to improve readability